pgjdbc/src/main/java/org/postgresql/util/PGServiceFile.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2018, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.util;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.text.MessageFormat;
+import java.util.HashMap;
+
+/*
+ * See https://www.postgresql.org/docs/current/static/libpq-pgservice.html.
+ */
+public class PGServiceFile {
+  public static HashMap load(String service) throws Exception {
+    String filename = findPath();
+    PGServiceFile file = new PGServiceFile();
+    BufferedReader in = new BufferedReader(new FileReader(filename));
+    file.parse(in);
+    return file.getService(service);
+  }
+
+  private static String findPath() {
+    String filename = System.getProperty("postgres.pgservicefile");
```

Comment:
I like the idea of namspacing the system property but should probably be prefixed with something like `org.postgresql.jdbc" so that it's pgjdbc specific rather than just "postgres.` (or `postgresql.`)

Reply:
I agree this should at least be `postgresql`, like the jar file. 

@davecramer @vlsi what do you think of this ? Should we use `org.postgresql.pgservicefile` ? Is there other system properties in pgjdbc ?

Reply:
`ConnectionUtil` uses `System.getProperty("database")` but that looks like test code. I don't find any System property in jdbc.